components/gitpod-protocol/src/util/gitpod-host-url.ts

```diff
@@ -24,11 +24,10 @@ const workspaceUrlPrefixRegex = RegExp(`^(([0-9]{4,6}|debug)-)?${baseWorkspaceID
 export class GitpodHostUrl {
     readonly url: URL;
 
-    constructor(url: string) {
-        const urlParam = url as any;
+    constructor(urlParam: string | URL) {
```

Comment:
I don't remember why but I believe we removed union type here on purpose before. And the only client which can pass URL inside it is this class itself for `with` functions